stdsimd/arch/detect/os/aarch64.rs

```diff
@@ -0,0 +1,64 @@
+//! Run-time feature detection for Aarch64 on any OS that traps the mrs instruction.
+
+use arch::detect::Feature;
+use arch::detect::cache;
+
+/// Try to read the features from the system registers.
+///
+/// This will cause SIGILL when the current OS is not trapping the mrs instruction.
+pub(crate) fn detect_features() -> cache::Initializer {
+    let mut value = cache::Initializer::default();
+
+    {
+        let mut enable_feature = |f, enable| {
+            if enable {
+                value.set(f as u32);
+            }
+        };
+
+        // ID_AA64ISAR0_EL1 - Instruction Set Attribute Register 0
+        let aa64isar0: u64;
+        unsafe { asm!("mrs $0, S3_0_C0_C6_0" : "=r"(aa64isar0)); }
+
+        let aes = bits_shift(aa64isar0, 7, 4) >= 1;
+        let pmull = bits_shift(aa64isar0, 7, 4) >= 2;
+        let sha1 = bits_shift(aa64isar0, 11, 8) >= 1;
+        let sha2 = bits_shift(aa64isar0, 15, 12) >= 1;
+        enable_feature(Feature::pmull, pmull);
+        // Crypto is specified as AES + PMULL + SHA1 + SHA2 per LLVM/hosts.cpp
+        enable_feature(Feature::crypto, aes && pmull && sha1 && sha2);
+        enable_feature(Feature::lse, bits_shift(aa64isar0, 23, 20) >= 1);
+        enable_feature(Feature::crc, bits_shift(aa64isar0, 19, 16) >= 1);
```

Comment:
This says >= 1 but [these docs](http://infocenter.arm.com/help/index.jsp?topic=/com.arm.doc.100458_0200_00_en/lau1442496521966.html) indicate the value may need to be exactly one?

Reply:
With these registers, the semantics is: higher number == higher level of support of this feature (which includes all previous ones). So e.g. if there's a next level of CRC32 implemented in the future, that CPU will have this value as 2.

e.g. Zircon implements checks as `>=`: https://fuchsia.googlesource.com/zircon/+/master/kernel/arch/arm64/feature.cpp#204

Reply:
Ah ok, good to know! Mind adding a comment somewhere in this file about that? (doesn't have to be on every comparison)

Reply:
It would be good if the top level comments could contain links to zircon and every other place that was relevant when coming up with the implementation.

Reply:
You have some links in the PR message, please consider adding all that information to the top of the module to avoid people from having to find the commit / PR to get it.